tools/cargo-miden/src/target.rs

```diff
@@ -6,6 +6,62 @@ use std::{
 
 use anyhow::{bail, Result};
 
+/// Represents whether the Cargo project is a Miden program or a library.
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum ProjectType {
+    Program,
+    Library,
+}
+
+/// Represents different types of rollup components that can be targeted
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum RollupTarget {
```

Comment:
This should probably go in `midenc_session` as part of the `TargetEnv` struct, no?

Reply:
Yes. I added it for future cases, and we don't distinguish account vs. note scripts for now. I'm hesitant to move it to `TargetEnv::Rollup` variant, since I'm not sure what is the best way to represent it in the CLI arguments. So, I removed `RollupTarget` for now and will get back to it when the need to distinguish account vs. note scripts arises.

Reply:
Oops, I meant `ProjectType`, not `TargetEnv`, the latter is just meant to distinguish between pure Rust and Rollup environments, it's the project type that will have the greatest affect on how we compile a project. Seems fine to me to come back to that though.